test/crud-with-options.test.js

```diff
@@ -536,6 +536,54 @@ describe('crud-with-options', function() {
 
   });
 
+  describe('upsert with where', function() {
+
+    beforeEach(seed);
+
+    it('should allow upsertWithWhere(data, cb)', function(done) {
+      User.upsertWithWhere({ name: 'John Smith' }, function() {
+        User.find({ where: { name: 'John Lennon' }}, function(err, data) {
+          should.not.exist(err);
+          data.length.should.equal(1);
+          User.find({ where: { name: 'John Smith' }}, function(err, data) {
+            should.not.exist(err);
+            data.length.should.equal(0);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should allow upsertWithWhere(where, data, cb)', function(done) {
+      User.upsertWithWhere({ name: 'John Lennon' }, { name: 'John Smith' }, function(err) {
+        should.not.exist(err);
+        User.find({ where: { name: 'John Lennon' }}, function(err, data) {
+          should.not.exist(err);
+          data.length.should.equal(0);
+          User.find({ where: { name: 'John Smith' }}, function(err, data) {
+            should.not.exist(err);
+            data.length.should.equal(1);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should allow upsertWithWhere ' +
+      '(where, data, options, cb)', function(done) {
+      options = undefined;
+      User.upsertWithWhere({ name: 'John Lennon' }, { name: 'John Smith'  }, options,
+        function(err) {
+          should.not.exist(err);
+          User.find({ where: { name: 'John Smith' }}, function(err, data) {
+            should.not.exist(err);
+            data.length.should.equal(1);
+            done();
+          });
+        });
+    });
+  });
+
```

Comment:
I guess it will be ok if we have tests just in `test/manipulation.test.js ` and `test/persistence-hooks.suite.js` if we have the full test coverage